### ISSUE TITLE
Merge fork-kde-master

### ISSUE THREAD

**Issue by ChALkeR** [qmlweb/qmlweb#19]
Merge kde/master by @akreuzkamp.

- [x] a24ef6c Add .gitignore — **Merged** in #29.
- [x] 9d4c4cf Don't override inner component context. — **Merged** in #56.
- [x] 2865c68 Allow creation of QtObject. — QtObject creation **merged** in #122. The rest of the changes **merged** in #161.
- [x] 08a4558 Properly remove animations at overwriting on init. — **Merged** in #152.
- [x] 486cc05 Use proper markdown syntax for README.md — **Rejected**, not required, inapplicable.
- [x] cbf467b Rename README to README.md — **Rejected**, not required, already done.
- [x] c42386b Improve README, update LICENSE, add a CONTRIBUTING — **Rejected**, README and CONTRIBUTING do not fit here. LICENSE has only year updates (already outdated), and that would be sorted out in #52. Please let me know if this is inappropriate.
- [x] 44bea45 Fix parsing of arrays containing bindings. — **Merged** in #57
- [x] f401c53 On parse error show extraxt of related source code — **Merged** in #57
- [x] 7dad8e8 Parser: Allow arrays with bindings. — **Merged** in #57
- [x] 8c6ec04 Add echoMode property to TextInput. — **Rejected**, already done in master, the remaining bits are at #47.
- [x] 2a7202f2 Add a custom implementation of QtWebChannel. — **Rejected**, the code here is LGPL, not BSD.
- [x] 79a05ff Add basic support for QtWebChannel — **Merged** in #57.
- [ ] 6022424 Add support for attached properties —  Attached properties **merged** in #160. Inheritance **merged** in #28.
- [x] 48db468 Remove double calling QMLBaseObject constructor — **Merged** in #29.
- [x] 2144c04 In $draw exit if mode!=Canvas instead of mode==DOM — **Rejected**, Canvas backend is dropped, inapplicable.
- [x] 4c9e432 Fix implicitSize of root element != body — **Merged** in #29.
- [x] 12b725e Add method Qt.createComponent similar to Qt's one — **Merged** in #29.
- [x] 11164f0 Fix some code alignment. — **Rejected**, code style only change, diverged too much, irrelevant now.
- [x] c3f1f39 Remove some old, unused helper functions — **Merged** in #29.
- [x] d35f213 Add a method nameForObject to root context object — **Merged** in #29.
- [x] 1c12ea4 Fix parsing of simple javascript-arrays — **Merged** in #29.
- [x] 3fab595 Rename TextArea to TextEdit to conform QtQuick. — **Merged** in #28, see also #48.
- [x] 785965c Make rootObject public via engine.rootObject — **Merged** in #29.
- [x] 3da78c7 Implement handler for JSItemModel::dataChanged — **Merged** in #29.
- [x] 02822c9 Fix parsing of files with windows line-feeds — **Merged** in #29.
- [x] b73fd64 Parse import statements correctly — **Merged** in #24.

**Comment by Plaristote** [comment]
I took care of b73fd64 in pull request #24.

**Comment by Plaristote** [comment]
11164f0 is irrelevant now, since all that code was already re-aligned in qt.js.
3fab595 should probably be done another way, since QML elements aren't described the way they used to be.
8c6ec04 though it is slightly different, this task is already implemented in master.

Otherwise, I made a new pull request #29 which takes care of 02822c9, 3da78c7, 785965c, 1c12ea4, d35f213, c3f1f39, 12b725e, 4c9e432, 48db468 and a24ef6c.
Basically everything from 02822c9 to 48db468, plus the last .gitignore commit (a24ef6c). The next commits might require more work.

**Comment by ChALkeR** [comment]
@Plaristote Great work! I updated the list above.

**Comment by henrikrudstrom** [comment]
merged 9d4c4cf14fded5c4dcbc499fb7f4d862a66a988e in #55

**Comment by ChALkeR** [comment]
@henrikrudstrom Could you do that in a separate PR, please, and keep the original author?

**Comment by ChALkeR** [comment]
Only 4 commits left here!

**Comment by ChALkeR** [comment]
I will create a PR today to review the remaining commits here.

**Comment by Plaristote** [comment]
I created PR #152 in order to merge 08a4558.

**Comment by ChALkeR** [comment]
I opened #160 for 6022424 and  #161 for 2865c68.

I guess those three PRs cover everything that is left here =).

**Comment by ChALkeR** [comment]
One commit left! :smile:

**Comment by ChALkeR** [comment]
Yay, we got everything from kde master merged now.

**Comment by Plaristote** [comment]
Sorry :( ! Something went wrong when I merged #160... unfortunately, I noticed it after rebasing, and the PR was considered merge... I re-opened it as #176. We first have to fix some tests.

**Comment by Plaristote** [comment]
Alright, I was able to re-close thanks to that comment: https://github.com/qmlweb/qmlweb/pull/160#issuecomment-209863745 !
